### ISSUE TITLE
Some search results are totally incorrect

### ISSUE THREAD

**Issue by piiskop** [osm-search/Nominatim#1877]
Try the coordinates 59.39736,24.29106 and see where the marker will be placed! That place is like a kilometer away from the correct place. I used your application via _Nextcloud Maps_ for an adventure game and had to reroute so many times because of incorrect results. I obtained correct results from Duckduckgo.

**Comment by mtmail** [comment]
Thanks for the detailed write-up on https://medium.com/arvutiteadus/foss-driving-directions-in-nextcloud-5ff23a87f83c?postPublishedType=repub And for comparing us to Apple Maps.

Nominatim returns the closest named object. http://nominatim.org/release-docs/latest/api/Reverse/ (the /search endpoint recognizes a coordinate query and act very similar).

https://nominatim.openstreetmap.org/reverse.php?format=html&lat=59.39736&lon=24.29106&zoom=20 (red small circle: search coordinates, blue larger circle: result). You might need to zoom out to see the red circle.

You're arguing it should return the search coordinates if there's no nearby object and use the name of the next higher encapsulating area, e.g. the forest https://www.openstreetmap.org/way/220023139#map=15/59.4081/24.3008 ?

It's possible to setup Nominatim to import all objects, or at least all streets. Then the result will be something like "unnamed bridge, Meremõisa küla, Lääne-Harju vald, Harju maakond, 76705, Estonia" https://www.openstreetmap.org/way/545690162, or "unnamed bridge, Keila-Joa alevik, Lääne-Harju vald, Harju maakond, 76701, Estonia" since the bridge is right between those.

For an OSRM navigation usecase it makes sense to import even the unamed streets. Or to give the cooridantes directly to OSRM. On nominatim.openstreetmap.org for a world wide installation it's too much data to handle.

**Comment by piiskop** [comment]
I copared it to /Google Maps/, not /Apple Maps/ since I don't use 
/Apple/ products at all and also want to get rid of using /Google/ products.

Thank you for explaining how Nominatim works! Now I see that there are 
two circles. However, /Nextcloud Maps/ only shows one marker. Why do you 
so eagerly want the location to have a name? There could be locations 
like a certain hole in a desert that I dug yesterday and I want to get 
the shortest route from there to another hole I dug tomorrow using real 
paths - both determined by coordinates. Google Maps can do that although 
they don't always start or finish by that location - the location is 
still visible. Names cause problems in your application as I reported in 
another issue where I handle the case about file naming - it's not 
possible to export a route with many intermediate stops. Of course, 
there are use cases where the name of the location is important. Could 
you just add an option for showing the closest named location for that? 
Unless there's such an option, /Nextcloud Maps/ is pretty much useless 
using search with coordinates. If I enter some coordinates, it's obvious 
that the location is determined by coordinates and not a name and the 
more digits the coordinates have the more precise the location of the 
marker must be.

So just put the marker to exact location and get rid of names and 
addresses if coordinates are entered!

*peacecop kalmer:*

*Instituudi tee 3-27 *

*76902 * *Harku *

*Harku*

*Harjumaa*
Estonia

*+372 652 4228 *

*+372  5620 4556 *

*__________________________________________________________________ *

**Comment by mtmail** [comment]
> I copared it to /Google Maps/, not /Apple Maps/ since I don't use /Apple/ products at all 

In the [blog post](https://medium.com/arvutiteadus/foss-driving-directions-in-nextcloud-5ff23a87f83c) you compare nominatim.openstreetmap.org results to https://duckduckgo.com/local.js results. Duckduckgo uses the commercial Apple Maps product https://spreadprivacy.com/duckduckgo-apple-mapkit-js/ 

Nominatim converts addresses to coordinates and coordinates to addresses. If you, or the maps/routing application you use, already know the precise coordinates you probably don't need a geocoding solution.

> Could you just add an option for showing the closest named location for that?

Developers or companies can install their own Nominatim database and choose to import unnamed streets. There's no plans to do that on nominatim.openstreetmap.org.

> as I reported in another issue where I handle the case about file naming - it's not possible to export a route [...]

Closing as duplicate of https://github.com/nextcloud/maps/issues/419 as it appears you mix up projects.

I'm impressed how patient other projects have been with your recent comments, or rather the tone how you write. "Are you a developer or just somebody that guesses? Developers should know exactly what they do. I see that you've contributed something to this project, so you should know, not guess." is something that would get deleted and user blocked on this issue tracker.
